app/src/main/java/de/xikolo/views/ExoPlayerVideoView.kt

```diff
@@ -0,0 +1,162 @@
+package de.xikolo.views
+
+import android.content.Context
+import android.net.Uri
+import android.support.annotation.IntRange
+import android.util.AttributeSet
+
+import com.google.android.exoplayer2.*
+import com.google.android.exoplayer2.source.ExtractorMediaSource
+import com.google.android.exoplayer2.source.MediaSource
+import com.google.android.exoplayer2.source.TrackGroupArray
+import com.google.android.exoplayer2.trackselection.AdaptiveTrackSelection
+import com.google.android.exoplayer2.trackselection.DefaultTrackSelector
+import com.google.android.exoplayer2.trackselection.TrackSelectionArray
+import com.google.android.exoplayer2.ui.PlayerView
+import com.google.android.exoplayer2.upstream.DefaultBandwidthMeter
+import com.google.android.exoplayer2.upstream.DefaultDataSourceFactory
+import com.google.android.exoplayer2.util.Util
+
+open class ExoPlayerVideoView : PlayerView {
+
+    private lateinit var mContext: Context
+    private lateinit var mPlayer: SimpleExoPlayer
+    private lateinit var mPlayerListener: Player.EventListener
+    private val mBandwidthMeter: DefaultBandwidthMeter = DefaultBandwidthMeter()
+    private var mMediaSource: MediaSource? = null
```

Comment:
[Just Say mNo to Hungarian Notation](https://jakewharton.com/just-say-no-to-hungarian-notation/)